test/jest/default-request-executor.test.js

```diff
@@ -0,0 +1,275 @@
+      };
+      const mockResponse = buildMockResponse({
+        headers: {
+          'x-okta-request-id' : 'foo'
+        }
+      });
+      const newRequest = requestExecutor.buildRetryRequest(mockRequest, mockResponse);
+      expect(newRequest.startTime).toEqual(startTime);
+    });
+  });
+
+  describe('canRetryRequest', () => {
+
+    it('should return true if X-Rate-Limit-Reset exists', () => {
+      const requestExecutor = new DefaultRequestExecutor();
+      const mockResponse = buildMockResponse({
+        headers: {
+          'x-rate-limit-reset': String(new Date())
+        }
+      });
+      expect(requestExecutor.canRetryRequest(mockResponse)).toBe(true);
+    });
+
+    it('should return false if X-Rate-Limit-Reset is absent', () => {
+      const requestExecutor = new DefaultRequestExecutor();
+      const mockResponse = buildMockResponse();
+      expect(requestExecutor.canRetryRequest(mockResponse)).toBe(false);
+    });
+
+    it('should return false if X-Rate-Limit-Reset exists twice ', () => {
+      const requestExecutor = new DefaultRequestExecutor();
+      const mockResponse = buildMockResponse({
+        headers: {
+          'x-rate-limit-reset': [String(new Date()), String(new Date())].join(',')
+        }
+      });
+      expect(requestExecutor.canRetryRequest(mockResponse)).toBe(false);
+    });
+  });
+
+  describe('getRetryDelayMs', () => {
+
+    it('should return a retry delay that is one second longer than the value set in the X-Rate-Limit-Reset header', () => {
+      const requestExecutor = new DefaultRequestExecutor();
+      const now = new Date();
+      const retryAt = new Date(now.getTime() + (1000 * 60));// one minute in the future
+      const mockResponse = buildMockResponse({
+        headers: {
+          'x-rate-limit-reset' : String(requestExecutor.dateToEpochSeconds(retryAt)),
+          date: now.toGMTString()
+        }
+      });
+      const delayMs = requestExecutor.getRetryDelayMs(mockResponse);
+      const delayDate = new Date(now.getTime() + delayMs);
+      expect(delayDate.getTime()).toBe(retryAt.getTime() + 1000);
+    });
+  });
+
+  describe('parseResponse', () => {
+
+    let requestExecutor;
+
+    beforeEach(() => {
+      requestExecutor = new DefaultRequestExecutor();
+      requestExecutor.retryRequest = jest.fn();
+    });
+
+    it('should defer to delayFetch for retry-able 429 responses, within max elapsed time', () => {
+      const uri = '/foo';
+      const request = { method: 'GET', uri };
+      const mockResponse = buildMockResponse({
+        status: 429,
+        headers: {
+          'x-rate-limit-reset': String(new Date())
+        }
+      });
+      requestExecutor.parseResponse(request, mockResponse);
+      expect(requestExecutor.retryRequest.mock.calls[0][0]).toBe(request);
+      expect(requestExecutor.retryRequest.mock.calls[0][1]).toBe(mockResponse);
+    });
+
+    it('should return 429 responses if max elapsed has passed', async () => {
```

Comment:
Since the default requestTimeout is now 0, we should a) have a test here to say that the default will continue to retry and b) have another test here that does what this is doing when a requestTimeout is specified.